### PR TITLE
fix(aapd-816): add max number validation

### PR DESCRIPTION
**Pull request by ms-ek**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-816

## Description of change

Added max nimber validation. Maximum number is 999.

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/forms-web-app/src/dynamic-forms/questions.js [hunk 1]
```diff
@@ -370,8 +370,9 @@ exports.questions = {
 			new ConditionalRequiredValidator('Enter how many days you expect the inquiry to last'),
 			new StringValidator({
 				regex: {
-					regex: '^(\\s*|\\d+)$',
-					regexMessage: 'How many days you expect the inquiry to last must be a number, like 30'
+					regex: '^([1-9]|[1-9]\\d{1,2})$',
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/questions.js:373]: `^\d{1,3}$`?

Comment (ms-ek) [reply packages/forms-web-app/src/dynamic-forms/questions.js:373]: Not too good with regex aha will try that

Comment (ms-ek) [reply packages/forms-web-app/src/dynamic-forms/questions.js:373]: That one accepts leading zeros, do you know how to make it from 1 to 999?

Comment (Si3rr4wow) [reply packages/forms-web-app/src/dynamic-forms/questions.js:373]: ooh I see yah, closer to your original one then `^[1-9]{1,3}$` chars from 1 to 9 between 1 and 3 times, does that work?

Comment (Si3rr4wow) [reply packages/forms-web-app/src/dynamic-forms/questions.js:373]: ah yeah this is awkward to validate as a string 😅

Comment (Si3rr4wow) [reply packages/forms-web-app/src/dynamic-forms/questions.js:373]: `^[1-9]\d{0,2}$` just a smidge simpler without the `|`

Screenshot 2023-11-13 at 11 21 03

Comment (ms-ek) [reply packages/forms-web-app/src/dynamic-forms/questions.js]: Nice that works thanks, updated the code

**Review by Si3rr4wow — APPROVED** [review]

**Pull request merged**